tests/user/test_get_user_by_id.py

```diff
@@ -0,0 +1,230 @@
+    @title("Getting User Info by ID with Invalid Token")
+    @description(
+        "GIVEN the user is registered, "
+        "WHEN the user sends a request to get user information by ID using an invalid token, "
+        "THEN the response code is 401 and the response body contains an appropriate error message."
+    )
+    def test_get_user_info_with_invalid_token(self, create_user):
+        with step("Getting user info by ID"):
+            user = create_user
+            invalid_token = "invalid_token"
+            getting_user_response = UsersAPI().get_user_by_id(token=invalid_token, user_id=user["id"])
+
+        with step("Checking the response code"):
+            assert_status_code(getting_user_response, 401)
+
+        with step("Checking the Content-Type"):
+            assert_content_type(getting_user_response, 'application/json')
+
+        with step("Checking the response body"):
+            expected_error_message = 'Internal server error'
+            assert_response_message(getting_user_response, expected_error_message)
+
+    @title("Getting User Info by ID with Non-Existing User ID")
+    @description(
+        "GIVEN the user is logged in, "
+        "WHEN the user sends a request to get information about non-existing user by ID, "
+        "THEN the response code is 404 and the response body contains an appropriate error message."
+    )
+    def test_get_user_info_with_non_existing_user_id(self, create_authorized_user):
+        user_id = '00a000a0-aa0a-0000-00a0-0000a00a0aaa'
+
+        with step("Getting user info by ID"):
+            token = create_authorized_user["token"]
+            getting_user_response = UsersAPI().get_user_by_id(user_id=user_id, token=token)
+
+        with step("Checking the response code"):
+            assert_status_code(getting_user_response, 404)
+
+        with step("Checking the Content-Type"):
+            assert_content_type(getting_user_response, 'application/json')
+
+        with step("Checking the response body"):
+            expected_error_message = f'User with id = {user_id} is not found.'
+            response = getting_user_response.json()
+            assert_response_message(getting_user_response, expected_error_message)
+            assert_that(response["httpStatusCode"], is_(404), reason='httpStatusCode should be 404')
+            assert_that(response["timestamp"], is_not(empty()), reason='timestamp should be present')
+
+    @pytest.mark.skip(reason="NEED TO CLARIFY: Bug in the API => wrong status code (403 instead of 404)")
+    @title("Getting User Info by ID with Invalid ID")
+    @description(
+        "GIVEN the user is logged in, "
+        "WHEN the user sends a request to get information about herself by invalid ID, "
+        "THEN the response code is 404 and the response body contains an appropriate error message."
+    )
+    def test_get_user_info_with_invalid_id(self, create_authorized_user):
```

Comment:
And `def test_get_user_info_with_invalid_id(self, create_authorized_user, invalid_id = '1234567890'):`?

Reply:
Combined in param test.